doc/langref.html.in

```diff
@@ -4683,6 +4705,35 @@ test "for reference" {
     try expect(items[2] == 3);
 }
 
+test "address of capture variable" {
+    var items = [_]i32 { 3, 4, 2 };
+    var all : [3]*const i32 = undefined;
+
+    // The following example has a bug.  After this loop executes, all contains
+    // 3 identical pointers, each pointing at the same i32, holding the last
+    // value iterated over.
+    // This happens because the value capture variable is per-loop, not
+    // per-iteration: &value is the same on every iteration, and value is
+    // overwritten on each iteration.
+    for (items, 0..) |value, i| {
+        all[i] = &value;
+    }
```

Comment:
Let us not give buggy examples except where such a bug would be caught by a compile error, or by runtime safety. These comments over-specify what happens, making promises about the language that are not true. For example, zig does not guarantee that in this code, `&value` is the same on every iteration.

Reply:
I removed the example; thanks for the review.
However I think this is a very subtle bug, as the experience with Go demonstrated (see https://github.com/golang/go/discussions/56010).

Another issue (assuming I'm interpreting your comment correctly) is that this example behavior is unspecified, but in the Zig Language Reference there is no documentation about *unspecified behavior*.